doc/rst/platforms/comm-layers/libfabric.rst

```diff
@@ -0,0 +1,319 @@
+structures.  When memory must be registered, the communication layer
+needs to know the maximum size the heap will grow to during execution.
+By default, the ofi communication layer creates a fixed heap whose size
+is 85% of compute node physical memory when it predicts that doing so
+will result in better network performance.
+
+You can adjust this by setting the
+``CHPL_RT_MAX_HEAP_SIZE`` environment variable.  Set it to a positive
+number for the desired heap size in bytes optionally followed by ``k``
+or ``K`` for KiB, ``m`` or ``M`` for MiB, ``g`` or ``G`` for GiB, or to
+a positive integer followed by ``%`` to indicate a percentage of the
+node real memory.  Either ``CHPL_RT_MAX_HEAP_SIZE=12g`` or ``=20%``
+specifies roughly a 12 GiB heap on a 64 GiB compute node, for example.
+
+Alternatively, you can prevent creation of a fixed heap entirely by
+setting ``CHPL_RT_MAX_HEAP_SIZE=0``.  This may cause the selection of a
+different provider than the highest-performing one, however.
+
+``CHPL_RT_MAX_HEAP_SIZE`` sets the heap size per compute node. If the
+application is run with multiple locales per node (co-locales), the heap size
+will be divided equally among the co-locales. For example, setting
+``CHPL_RT_MAX_HEAP_SIZE=100G`` and running with 4 co-locales (i.e., ``-nl Nx4``) will
+result in a 25 GiB heap per co-locale.
+
+.. note::
+  In the future we hope to be able to reduce the user impact of memory
+  registration and fixed heaps when using the ofi communication layer.
+
+
+Hugepages on Cray XC and HPE Cray EX Systems
+____________________________________________
+
+We have not yet quantified the effects, but performance with the gni and
+cxi providers may be improved through the use of hugepages. To use
+hugepages you must have a ``craype-hugepages`` module loaded and
+``CHPL_COMM_OFI_USE_HUGEPAGES=true`` when you build Chapel and compile
+your program, and you must have a ``craype-hugepages`` module loaded
+and ``CHPL_RT_COMM_OFI_USE_HUGEPAGES=true`` when you run your program
+(note that these are not the same variable.)
+
+For
+example::
+
+     module load craype-hugepages16M
+     CHPL_COMM_OFI_USE_HUGEPAGES=true
+     CHPL_RT_COMM_OFI_USE_HUGEPAGES=true
+
+See :ref:`ugni-comm-on-cray` for more discussion about hugepages,
+hugepage modules, and the heap size.  Note, however, that anything
+there about a dynamically sized heap does not apply to the ofi
+communication layer and the libfabric gni and cxi providers. Also note
+that the ofi communication layer cannot test the size of a hugepage
+fixed heap during initialization; as a result, it will create a heap of
+size ``CHPL_RT_MAX_HEAP_SIZE`` even if there is insufficient physical
+memory, which may lead to out-of-memory errors during program
+execution.
+
+
+Transparent Hugepages
+_____________________
+
+With some providers, the use of transparent hugepages is required to register
+more memory. This is default when using the EFA provider on AWS. To explicitly
```

Comment:
Not your fault, but this is missing a word: "This is *the* default"

Reply:
I wrote these docs over a year ago, it probably is my fault :smile: